test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```
@@ -335,82 +169,88 @@ protected Settings restClientSettings() {
+                try (InputStream is = Files.newInputStream(path)) {
+                    keyStore.load(is, keystorePass.toCharArray());
+                }
+                SSLContext sslcontext = SSLContexts.custom().loadTrustMaterial(keyStore, null).build();
+                SSLConnectionSocketFactory sslConnectionSocketFactory = new SSLConnectionSocketFactory(sslcontext);
+                builder.setHttpClientConfigCallback(new SSLSocketFactoryHttpConfigCallback(sslConnectionSocketFactory));
+            } catch (KeyStoreException|NoSuchAlgorithmException|KeyManagementException|CertificateException|IOException e) {
+                throw new RuntimeException(e);
             }
-            logger.debug("end setup test [{}]", testCandidate.getTestPath());
         }
 
-        restTestExecutionContext.clear();
-
         try {
-            for (ExecutableSection executableSection : testCandidate.getTestSection().getExecutableSections()) {
-                executableSection.execute(restTestExecutionContext);
-            }
-        } finally {
-            logger.debug("start teardown test [{}]", testCandidate.getTestPath());
-            for (DoSection doSection : testCandidate.getTeardownSection().getDoSections()) {
-                doSection.execute(restTestExecutionContext);
+            try (ThreadContext threadContext = new ThreadContext(settings)) {
+                Header[] defaultHeaders = new Header[threadContext.getHeaders().size()];
+                int i = 0;
+                for (Map.Entry<String, String> entry : threadContext.getHeaders().entrySet()) {
+                    defaultHeaders[i++] = new BasicHeader(entry.getKey(), entry.getValue());
+                }
+                builder.setDefaultHeaders(defaultHeaders);
             }
-            logger.debug("end teardown test [{}]", testCandidate.getTestPath());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

shouldn't we let the IOException bubble up?